tests/ts/examples.ts

```diff
@@ -57,6 +27,54 @@ class Person extends objection.Model {
   }
 }
 
+const takesVoid = (v: void) => 1;
+
+const takesPerson = (person: Person) => {
+  person.examplePersonMethod("");
+};
+const takesMaybePerson = (_: Person | undefined) => 1;
```

Comment:
This just caught my eye, not sure if its wrong or not. 

Is it really necessary to explicitly allow also passing `undefined`? I thought that undefined would be always allowed (or would that be equivalent with `_?: Person`).

Reply:
I wanted to do type verification for methods that optionally return a Person. Both signatures would suffice, either ` (_: Person | undefined) => ...` or ` (_?: Person) =>...`.